src/include/homestore/logstore/log_store.hpp

```diff
@@ -231,6 +231,7 @@ class HomeLogStore : public std::enable_shared_from_this< HomeLogStore > {
     bool rollback(logstore_seq_num_t to_lsn);
 
     auto start_lsn() const { return m_start_lsn.load(std::memory_order_acquire); }
+    auto tail_lsn() const { return m_tail_lsn.load(std::memory_order_acquire); }
```

Comment:
so this line is the actual fix？
thedummy log is a right fix but we don't expect to hit?

Reply:
Yes, this line and` set_last_durable_lsn(hs->tail_lsn());` is the actual fix to avoid hitting 'dummy log at last_entry()'

Reply:
1 according the [nuraft logstore#last_entry()](https://github.com/eBay/NuRaft/blob/50e2f949503081262cb21923e633eaa8dacad8fa/include/libnuraft/log_store.hxx#L56), a dummy log should return If no log entry exists, and should not throw an exception. 

2 set_last_durable_lsn(hs->tail_lsn() makes sure HomeRaftLogStore#m_last_durable_lsn is set to up-to-date we restart. now , m_last_durable_lsn is only updated when log is flushed (end_of_log_batch), which means if we do not append any log after restart , the m_last_durable_lsn will be a default value(-1), which might cause a issue.

in sm_long running issue 19, after restart, no log entry is appended. and this node starts handing vote, where ogstore#last_entry() will be called, but at this moment , m_last_durable_lsn is -1 , which shuold be the last lsn in log store(tail_lsn). this will cause an error that log entry not found